gmni{,srv}: init at 2021-03-26

jb55 opened the pull request:
- gmni: init at 2021-03-26
- gmnisrv: init at 2021-03-26

SuperSandro2000 commented on pkgs/servers/gemini/gmnisrv/default.nix:
```diff
@@ -0,0 +1,22 @@
+  };
+  MIMEDB = "${mailcap}/etc/mime.types";
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl mailcap scdoc ];
+  meta = with lib; {
```
```suggestion
  buildInputs = [ openssl mailcap scdoc ];

  meta = with lib; {
```

SuperSandro2000 commented on pkgs/servers/gemini/gmnisrv/default.nix:
```diff
@@ -0,0 +1,22 @@
+  meta = with lib; {
+    description = "A simple Gemini protocol server";
+    homepage = "https://git.sr.ht/~sircmpwn/gmnisrv";
+    license = licenses.gpl3;
```
gpl3Only or gpl3Plus?

SuperSandro2000 commented on pkgs/servers/gemini/gmnisrv/default.nix:
```diff
@@ -0,0 +1,22 @@
+
+stdenv.mkDerivation rec {
+  name = "gmnvisrv";
+  version = "2021-03-26";
+  src = fetchFromSourcehut {
```
```suggestion
  version = "unstable-2021-03-26";

  src = fetchFromSourcehut {
```

SuperSandro2000 commented on pkgs/servers/gemini/gmnisrv/default.nix:
```diff
@@ -0,0 +1,22 @@
+    repo = "gmnisrv";
+    rev = "f23ec10a6d66c574bbf718c4b10f2cf91ea8daef";
+    sha256 = "1n0iiz6gd3dmv1r7i6437zmrv0vim69fc0haqfvagd4y20hy5s8z";
+  };
+  MIMEDB = "${mailcap}/etc/mime.types";
```
```suggestion
  };

  MIMEDB = "${mailcap}/etc/mime.types";
```

SuperSandro2000 commented on pkgs/applications/networking/browsers/gmni/default.nix:
```diff
@@ -0,0 +1,21 @@
+  meta = with lib; {
+    description = "A Gemini client";
+    homepage = "https://git.sr.ht/~sircmpwn/gmni";
+    license = licenses.gpl3;
```
gpl3Only or gpl3Plus?

SuperSandro2000 commented on pkgs/applications/networking/browsers/gmni/default.nix:
```diff
@@ -0,0 +1,21 @@
+    sha256 = "1wvnzyv7vyddcd39y6q5aflpnnsdl4k4y5aj5ssb7vgkld0h1b7r";
+  };
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ bearssl scdoc ];
+  meta = with lib; {
```
```suggestion
  buildInputs = [ bearssl scdoc ];

  meta = with lib; {
```

SuperSandro2000 commented on pkgs/applications/networking/browsers/gmni/default.nix:
```diff
@@ -0,0 +1,21 @@
+    repo = "gmni";
+    rev = "77b73efbcd3ea7ed9e3e4c0aa19d9247e21d3c87";
+    sha256 = "1wvnzyv7vyddcd39y6q5aflpnnsdl4k4y5aj5ssb7vgkld0h1b7r";
+  };
+  nativeBuildInputs = [ pkg-config ];
```
```suggestion
  };

  nativeBuildInputs = [ pkg-config ];
```

SuperSandro2000 commented on pkgs/applications/networking/browsers/gmni/default.nix:
```diff
@@ -0,0 +1,21 @@
+
+stdenv.mkDerivation rec {
+  name = "gmni";
+  version = "2021-03-26";
+  src = fetchFromSourcehut {
```
```suggestion
  version = "unstable-2021-03-26";

  src = fetchFromSourcehut {
```

SuperSandro2000 commented on pkgs/applications/networking/browsers/gmni/default.nix:
```diff
@@ -0,0 +1,21 @@
+{stdenv, lib, fetchFromSourcehut, pkg-config, bearssl, scdoc}:
```
```suggestion
{ stdenv, lib, fetchFromSourcehut, pkg-config, bearssl, scdoc }:
```

SuperSandro2000 commented on pkgs/servers/gemini/gmnisrv/default.nix:
```diff
@@ -0,0 +1,22 @@
+{stdenv, lib, fetchFromSourcehut, pkg-config, openssl, mailcap, scdoc}:
```
```suggestion
{ stdenv, lib, fetchFromSourcehut, pkg-config, openssl, mailcap, scdoc }:
```

SuperSandro2000 reviewed (changes requested):
if you are using mailcap just for mime-types it can be replaced with https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/misc/mime-types/default.nix.

r-rmcgibbo commented:
Result of `nixpkgs-review pr 118056` at 43956f38 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/48f42016ba01d162075c79e2b8795c52">gmnisrv</a></li>
  </ul>
</details>
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>gmni</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/servers/gemini/gmnisrv/default.nix:18:5:
```
   |
18 |     license = licenses.gpl3;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/applications/networking/browsers/gmni/default.nix:17:5:
```
   |
17 |     license = licenses.gpl3;
   |     ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

bsima commented:
> 1 package failed to build:
>   - gmnisrv

Idk why this is failing.... works on my machine...

```
hash mismatch in fixed-output derivation '/nix/store/3x2sr2197wl8ni9bmgaf33lnj3jmaahj-source':
  wanted: sha256:1n0iiz6gd3dmv1r7i6437zmrv0vim69fc0haqfvagd4y20hy5s8z
  got:    sha256:1d9rjx0s092yfzjxd2yvzixhqgg883nlnmsysgp21w75n2as354n
```

Both of these sha's work for me... I seem to recall that there are some weird bugs with the nix tarball cache, and maybe this is an instance of such?

Anyway I'll use the `1d9rjx` sha that the build server wants.

bsima committed:
gmni: init at unstable-2021-03-26

Reviewed-by: William Casarin <jb55@jb55.com>
Link: https://lists.sr.ht/~andir/nixpkgs-dev/%3C20210330145338.8149-1-ben@bsima.me%3E

bsima committed:
gmnisrv: init at unstable-2021-03-26

Reviewed-by: William Casarin <jb55@jb55.com>
Link: https://lists.sr.ht/~andir/nixpkgs-dev/%3C20210330145338.8149-2-ben@bsima.me%3E

SuperSandro2000 commented:
> Both of these sha's work for me... I seem to recall that there are some weird bugs with the nix tarball cache, and maybe this is an instance of such?

This is not a bug but a feature. You have locally cached both tarballs but others cannot reproduce them.

SuperSandro2000 commented:
> I think gpl3Only? Really not sure what the differences are, the texts of each license on spdx.org look exactly the same to me...

if the project declares somewhere ``or later``. Most likely this is in source or in the readme, not the license itself.

SuperSandro2000 commented on pkgs/servers/gemini/gmnisrv/default.nix:
```diff
@@ -0,0 +1,25 @@
+{ stdenv, lib, fetchFromSourcehut, pkg-config, openssl, mime-types, scdoc }:
+
+stdenv.mkDerivation rec {
+  name = "gmnvisrv";
```
```suggestion
  pname = "gmnvisrv";
```

SuperSandro2000 commented on pkgs/applications/networking/browsers/gmni/default.nix:
```diff
@@ -0,0 +1,24 @@
+{ stdenv, lib, fetchFromSourcehut, pkg-config, bearssl, scdoc }:
+
+stdenv.mkDerivation rec {
+  name = "gmni";
```
```suggestion
  pname = "gmni";
```

jb55 commented:
On Wed, Mar 31, 2021 at 10:37:33AM -0700, Sandro wrote:
>@SuperSandro2000 requested changes on this pull request.
>  pname = "gmnvisrv";
>  pname = "gmni";

good catch, thanks. pushed a fix for these

bsima commented on pkgs/servers/gemini/gmnisrv/default.nix:
```diff
@@ -0,0 +1,25 @@
+{ stdenv, lib, fetchFromSourcehut, pkg-config, openssl, mime-types, scdoc }:
+
+stdenv.mkDerivation rec {
+  pname = "gmnvisrv";
```
ah I typo'd, there's an extra `v`

```diff
-gmnvisrv
+gmnisrv
```

jb55 commented:
On Wed, Mar 31, 2021 at 04:11:49PM -0700, Ben Sima wrote:
***@***.*** commented on this pull request.
>+stdenv.mkDerivation rec {
>+  pname = "gmnvisrv";

Might have been me in last amend. Fixed.
